File: KahveQRKodOlusturucuJava-calismalarim/app/src/main/java/com/example/qrkodolusturucujava/model/UrunKategori.java
package com.example.qrkodolusturucujava.model;

public enum UrunKategori {
    INDIRIMLI,
    ATISTIRMALIKLAR,
    ICECEKLER;

    public int kategoriKodu() {
        switch (this) {
            case INDIRIMLI:
                return 0;
            case ATISTIRMALIKLAR:
                return 1;
            case ICECEKLER:
                return 2;
            default:
                return -1;
        }
    }
}

File: KahveQrOlusturucuKotlin-calismalarim/app/src/main/java/com/example/qrkodolusturucu/model/UrunKategori.kt
package com.example.qrkodolusturucu.model

enum class UrunKategori {
    INDIRIMLI, ATISTIRMALIKLAR, ICECEKLER;

    fun kategoriKodu(): Int {
        return when (this) {
            INDIRIMLI -> 0
            ATISTIRMALIKLAR -> 1
            ICECEKLER -> 2
        }
    }
}